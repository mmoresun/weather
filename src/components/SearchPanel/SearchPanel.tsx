
import TextField from "@mui/material/TextField";
import Button from "@mui/material/Button";
import Box from '@mui/material/Box';

const SearchPanel = ({ getCity }: { getCity: any }) => {
    return (
        // <div style={{ margin: '20px 0 0 0' }}>
        <Box
            component="form"
            onSubmit={() => getCity}
        >
            <TextField
                size='small'
                type="text"
                name='city'
                placeholder='Enter city'
            />
            <Button

                type="submit"
            >
                Get weather
            </Button>
        </Box>
        // </div >
    );
}

export default SearchPanel;

