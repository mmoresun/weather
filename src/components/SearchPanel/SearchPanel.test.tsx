import SearchPanel from "./SearchPanel";
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

it('renders SearchPanel correctly', () => {
  const tree = renderer
    .create(<SearchPanel />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders search form', () => {
  render(<SearchPanel />);
  // const searchForm = screen.getByTitle('searchpanel')
  const searchForm = screen.getByRole('form');
  expect(searchForm).toBeInTheDocument();
});

it('renders SearchPanel input field', () => {
  render(<SearchPanel />);
  const searchInput = screen.getByPlaceholderText('Enter city');
  expect(searchInput).toBeInTheDocument();
});

it('renders SearchPanel search button', () => {
  render(<SearchPanel />);
  const searchButton = screen.getByRole('button');
  expect(searchButton).toBeInTheDocument();
});

