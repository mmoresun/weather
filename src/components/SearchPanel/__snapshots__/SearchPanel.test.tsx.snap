// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SearchPanel correctly 1`] = `
<form
  className="MuiBox-root css-0"
  name="searchpanel"
>
  <div
    className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id=":r0:"
        name="city"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Enter city"
        required={false}
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-ihdtdm"
        >
          <span
            className="notranslate"
          >
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    Get weather
  </button>
</form>
`;
